<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
<meta http-equiv="x-ua-compatible" content="IE=edge"/>
<title>Test results - Class com.stehno.ersatz.impl.DigestSpec</title>
<link href="../css/base-style.css" rel="stylesheet" type="text/css"/>
<link href="../css/style.css" rel="stylesheet" type="text/css"/>
<script src="../js/report.js" type="text/javascript"></script>
</head>
<body>
<div id="content">
<h1>Class com.stehno.ersatz.impl.DigestSpec</h1>
<div class="breadcrumbs">
<a href="../index.html">all</a> &gt; 
<a href="../packages/com.stehno.ersatz.impl.html">com.stehno.ersatz.impl</a> &gt; DigestSpec</div>
<div id="summary">
<table>
<tr>
<td>
<div class="summaryGroup">
<table>
<tr>
<td>
<div class="infoBox" id="tests">
<div class="counter">2</div>
<p>tests</p>
</div>
</td>
<td>
<div class="infoBox" id="failures">
<div class="counter">0</div>
<p>failures</p>
</div>
</td>
<td>
<div class="infoBox" id="ignored">
<div class="counter">0</div>
<p>ignored</p>
</div>
</td>
<td>
<div class="infoBox" id="duration">
<div class="counter">1.109s</div>
<p>duration</p>
</div>
</td>
</tr>
</table>
</div>
</td>
<td>
<div class="infoBox success" id="successRate">
<div class="percent">100%</div>
<p>successful</p>
</div>
</td>
</tr>
</table>
</div>
<div id="tabs">
<ul class="tabLinks">
<li>
<a href="#tab0">Tests</a>
</li>
<li>
<a href="#tab1">Standard output</a>
</li>
<li>
<a href="#tab2">Standard error</a>
</li>
</ul>
<div id="tab0" class="tab">
<h2>Tests</h2>
<table>
<thead>
<tr>
<th>Test</th>
<th>Duration</th>
<th>Result</th>
</tr>
</thead>
<tr>
<td class="success">DIGEST auth</td>
<td class="success">0.286s</td>
<td class="success">passed</td>
</tr>
<tr>
<td class="success">DIGEST no-auth</td>
<td class="success">0.823s</td>
<td class="success">passed</td>
</tr>
</table>
</div>
<div id="tab1" class="tab">
<h2>Standard output</h2>
<span class="code">
<pre>06:32:01.626 [Test worker] DEBUG io.undertow - starting undertow server io.undertow.Undertow@77876ddc
06:32:01.652 [XNIO-2 I/O-1] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 I/O-1', selector sun.nio.ch.EPollSelectorImpl@3e08b362
06:32:01.680 [XNIO-2 I/O-2] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 I/O-2', selector sun.nio.ch.EPollSelectorImpl@58f8538c
06:32:01.680 [XNIO-2 I/O-3] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 I/O-3', selector sun.nio.ch.EPollSelectorImpl@6841a6aa
06:32:01.697 [XNIO-2 I/O-4] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 I/O-4', selector sun.nio.ch.EPollSelectorImpl@396e569b
06:32:01.716 [XNIO-2 I/O-5] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 I/O-5', selector sun.nio.ch.EPollSelectorImpl@583094c1
06:32:01.730 [XNIO-2 I/O-6] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 I/O-6', selector sun.nio.ch.EPollSelectorImpl@73d6233c
06:32:01.748 [XNIO-2 I/O-7] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 I/O-7', selector sun.nio.ch.EPollSelectorImpl@6c7510
06:32:01.748 [XNIO-2 I/O-8] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 I/O-8', selector sun.nio.ch.EPollSelectorImpl@23757e9a
06:32:01.772 [Test worker] DEBUG io.undertow - Configuring listener with protocol HTTP for interface localhost and port 0
06:32:01.772 [XNIO-2 Accept] DEBUG org.xnio.nio - Started channel thread 'XNIO-2 Accept', selector sun.nio.ch.EPollSelectorImpl@11f5ebe
06:32:02.090 [XNIO-2 task-1] DEBUG io.undertow.request.security - Setting authentication required for exchange HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:36287]} response {}}
06:32:02.091 [XNIO-2 task-1] DEBUG io.undertow.request.security - Attempting to authenticate /secrets, authentication required: true
06:32:02.091 [XNIO-2 task-1] DEBUG io.undertow.request.security - Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.DigestAuthenticationMechanism@5ae6b23f for /secrets
06:32:02.092 [XNIO-2 task-1] DEBUG io.undertow.request.security - Sending authentication challenge for HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:36287]} response {}}
06:32:02.107 [XNIO-2 task-1] DEBUG io.undertow.request.security - Authentication result was CHALLENGE_SENT for /secrets
06:32:02.156 [XNIO-2 task-2] DEBUG io.undertow.request.security - Setting authentication required for exchange HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Authorization=[Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;nLgTW0S//BYNMTUzMjA4NjMyMjEwNNVgdE/bD4+m5wTf8Zv+RNA=&quot;, uri=&quot;/secrets&quot;, response=&quot;37023761d024de17eb52f08e4c821907&quot;, qop=auth, nc=00000001, cnonce=&quot;c9c9319b3421520a&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot;], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:36287]} response {}}
06:32:02.159 [XNIO-2 task-2] DEBUG io.undertow.request.security - Attempting to authenticate /secrets, authentication required: true
06:32:02.193 [XNIO-2 task-2] DEBUG io.undertow.request.security - Found digest header Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;nLgTW0S//BYNMTUzMjA4NjMyMjEwNNVgdE/bD4+m5wTf8Zv+RNA=&quot;, uri=&quot;/secrets&quot;, response=&quot;37023761d024de17eb52f08e4c821907&quot;, qop=auth, nc=00000001, cnonce=&quot;c9c9319b3421520a&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot; in HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Authorization=[Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;nLgTW0S//BYNMTUzMjA4NjMyMjEwNNVgdE/bD4+m5wTf8Zv+RNA=&quot;, uri=&quot;/secrets&quot;, response=&quot;37023761d024de17eb52f08e4c821907&quot;, qop=auth, nc=00000001, cnonce=&quot;c9c9319b3421520a&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot;], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:36287]} response {}}
06:32:02.230 [XNIO-2 task-2] DEBUG io.undertow.request.security - Authentication failed with message UT000038: Authentication failed, requested user name 'admin' and mechanism DIGEST for HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Authorization=[Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;nLgTW0S//BYNMTUzMjA4NjMyMjEwNNVgdE/bD4+m5wTf8Zv+RNA=&quot;, uri=&quot;/secrets&quot;, response=&quot;37023761d024de17eb52f08e4c821907&quot;, qop=auth, nc=00000001, cnonce=&quot;c9c9319b3421520a&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot;], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:36287]} response {}}
06:32:02.231 [XNIO-2 task-2] DEBUG io.undertow.request.security - Authentication outcome was NOT_AUTHENTICATED with method io.undertow.security.impl.DigestAuthenticationMechanism@5ae6b23f for /secrets
06:32:02.231 [XNIO-2 task-2] DEBUG io.undertow.request.security - Sending authentication challenge for HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Authorization=[Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;nLgTW0S//BYNMTUzMjA4NjMyMjEwNNVgdE/bD4+m5wTf8Zv+RNA=&quot;, uri=&quot;/secrets&quot;, response=&quot;37023761d024de17eb52f08e4c821907&quot;, qop=auth, nc=00000001, cnonce=&quot;c9c9319b3421520a&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot;], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:36287]} response {}}
06:32:02.232 [XNIO-2 task-2] DEBUG io.undertow.request.security - Authentication result was CHALLENGE_SENT for /secrets
06:32:02.243 [Test worker] DEBUG io.undertow - stopping undertow server io.undertow.Undertow@77876ddc
06:32:02.283 [Test worker] DEBUG io.undertow - starting undertow server io.undertow.Undertow@61f50a7a
06:32:02.304 [XNIO-3 I/O-2] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 I/O-2', selector sun.nio.ch.EPollSelectorImpl@1159c326
06:32:02.304 [XNIO-3 I/O-1] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 I/O-1', selector sun.nio.ch.EPollSelectorImpl@21e37daf
06:32:02.328 [XNIO-3 I/O-3] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 I/O-3', selector sun.nio.ch.EPollSelectorImpl@4891422a
06:32:02.332 [XNIO-3 I/O-4] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 I/O-4', selector sun.nio.ch.EPollSelectorImpl@3afff3c7
06:32:02.360 [XNIO-3 I/O-5] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 I/O-5', selector sun.nio.ch.EPollSelectorImpl@70fe52f9
06:32:02.368 [XNIO-3 I/O-6] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 I/O-6', selector sun.nio.ch.EPollSelectorImpl@2ecdd125
06:32:02.380 [XNIO-3 I/O-7] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 I/O-7', selector sun.nio.ch.EPollSelectorImpl@681dec2d
06:32:02.408 [XNIO-3 I/O-8] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 I/O-8', selector sun.nio.ch.EPollSelectorImpl@5a6633a2
06:32:02.408 [Test worker] DEBUG io.undertow - Configuring listener with protocol HTTP for interface localhost and port 0
06:32:02.412 [XNIO-3 Accept] DEBUG org.xnio.nio - Started channel thread 'XNIO-3 Accept', selector sun.nio.ch.EPollSelectorImpl@4d891282
06:32:02.460 [XNIO-3 task-1] DEBUG io.undertow.request.security - Setting authentication required for exchange HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:41627]} response {}}
06:32:02.468 [XNIO-3 task-1] DEBUG io.undertow.request.security - Attempting to authenticate /secrets, authentication required: true
06:32:02.469 [XNIO-3 task-1] DEBUG io.undertow.request.security - Authentication outcome was NOT_ATTEMPTED with method io.undertow.security.impl.DigestAuthenticationMechanism@6b509983 for /secrets
06:32:02.469 [XNIO-3 task-1] DEBUG io.undertow.request.security - Sending authentication challenge for HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:41627]} response {}}
06:32:02.470 [XNIO-3 task-1] DEBUG io.undertow.request.security - Authentication result was CHALLENGE_SENT for /secrets
06:32:02.492 [XNIO-3 task-2] DEBUG io.undertow.request.security - Setting authentication required for exchange HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Authorization=[Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;aN/LaaGp8PQNMTUzMjA4NjMyMjQ3MEo8WFUGmu7vFibxA02s8yM=&quot;, uri=&quot;/secrets&quot;, response=&quot;5b32dbe034ece2602878e73e353ad284&quot;, qop=auth, nc=00000001, cnonce=&quot;ccc0ead5e83ba45b&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot;], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:41627]} response {}}
06:32:02.493 [XNIO-3 task-2] DEBUG io.undertow.request.security - Attempting to authenticate /secrets, authentication required: true
06:32:02.504 [XNIO-3 task-2] DEBUG io.undertow.request.security - Found digest header Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;aN/LaaGp8PQNMTUzMjA4NjMyMjQ3MEo8WFUGmu7vFibxA02s8yM=&quot;, uri=&quot;/secrets&quot;, response=&quot;5b32dbe034ece2602878e73e353ad284&quot;, qop=auth, nc=00000001, cnonce=&quot;ccc0ead5e83ba45b&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot; in HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Authorization=[Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;aN/LaaGp8PQNMTUzMjA4NjMyMjQ3MEo8WFUGmu7vFibxA02s8yM=&quot;, uri=&quot;/secrets&quot;, response=&quot;5b32dbe034ece2602878e73e353ad284&quot;, qop=auth, nc=00000001, cnonce=&quot;ccc0ead5e83ba45b&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot;], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:41627]} response {}}
06:32:02.509 [XNIO-3 task-2] DEBUG io.undertow.request.security - Authenticated as admin, roles [TESTER]
06:32:02.511 [XNIO-3 task-2] DEBUG io.undertow.request.security - Authentication outcome was AUTHENTICATED with method io.undertow.security.impl.DigestAuthenticationMechanism@6b509983 for /secrets
06:32:02.512 [XNIO-3 task-2] DEBUG io.undertow.request.security - Authentication result was AUTHENTICATED for /secrets
06:32:02.512 [XNIO-3 task-2] DEBUG com.stehno.ersatz.ErsatzServer - Request: { GET /secrets (query=[:], headers={Connection=[Keep-Alive], Authorization=[Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;aN/LaaGp8PQNMTUzMjA4NjMyMjQ3MEo8WFUGmu7vFibxA02s8yM=&quot;, uri=&quot;/secrets&quot;, response=&quot;5b32dbe034ece2602878e73e353ad284&quot;, qop=auth, nc=00000001, cnonce=&quot;ccc0ead5e83ba45b&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot;], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:41627]}, cookies=[:]): &lt;empty&gt; }
06:32:02.516 [XNIO-3 task-2] DEBUG com.stehno.ersatz.ErsatzServer - Response({Authentication-Info=[nextnonce=&quot;aN/LaaGp8PQNMTUzMjA4NjMyMjQ3MEo8WFUGmu7vFibxA02s8yM=&quot;,qop=&quot;auth&quot;,rspauth=&quot;86782f3ca3ccbc88647db572e6574961&quot;,cnonce=&quot;ccc0ead5e83ba45b&quot;,nc=00000001]}): Something secure
06:32:02.518 [Test worker] DEBUG io.undertow - stopping undertow server io.undertow.Undertow@61f50a7a
06:32:02.528 [XNIO-3 task-2] ERROR io.undertow.request - UT005071: Undertow request failed HttpServerExchange{ GET /secrets request {Connection=[Keep-Alive], Authorization=[Digest username=&quot;admin&quot;, realm=&quot;DigestTesting&quot;, nonce=&quot;aN/LaaGp8PQNMTUzMjA4NjMyMjQ3MEo8WFUGmu7vFibxA02s8yM=&quot;, uri=&quot;/secrets&quot;, response=&quot;5b32dbe034ece2602878e73e353ad284&quot;, qop=auth, nc=00000001, cnonce=&quot;ccc0ead5e83ba45b&quot;, algorithm=MD5, opaque=&quot;00000000000000000000000000000000&quot;], Accept-Encoding=[gzip], User-Agent=[okhttp/3.11.0], Host=[localhost:41627]} response {Content-Encoding=[gzip], Connection=[keep-alive], Authentication-Info=[nextnonce=&quot;aN/LaaGp8PQNMTUzMjA4NjMyMjQ3MEo8WFUGmu7vFibxA02s8yM=&quot;,qop=&quot;auth&quot;,rspauth=&quot;86782f3ca3ccbc88647db572e6574961&quot;,cnonce=&quot;ccc0ead5e83ba45b&quot;,nc=00000001], Content-Length=[36], Date=[Fri, 20 Jul 2018 11:32:01 GMT]}}
java.lang.NullPointerException: null
	at io.undertow.conduits.DeflatingStreamSinkConduit.performFlushIfRequired(DeflatingStreamSinkConduit.java:449)
	at io.undertow.conduits.DeflatingStreamSinkConduit.flush(DeflatingStreamSinkConduit.java:371)
	at org.xnio.conduits.ConduitStreamSinkChannel.flush(ConduitStreamSinkChannel.java:162)
	at io.undertow.channels.DetachableStreamSinkChannel.flush(DetachableStreamSinkChannel.java:119)
	at org.xnio.channels.Channels.flushBlocking(Channels.java:63)
	at io.undertow.io.UndertowOutputStream.close(UndertowOutputStream.java:347)
	at io.undertow.io.BlockingSenderImpl.close(BlockingSenderImpl.java:219)
	at io.undertow.io.DefaultIoCallback.onComplete(DefaultIoCallback.java:54)
	at io.undertow.io.BlockingSenderImpl.invokeOnComplete(BlockingSenderImpl.java:276)
	at io.undertow.io.BlockingSenderImpl.send(BlockingSenderImpl.java:132)
	at io.undertow.io.BlockingSenderImpl.send(BlockingSenderImpl.java:166)
	at com.stehno.ersatz.ErsatzServer.send(ErsatzServer.groovy:539)
	at com.stehno.ersatz.ErsatzServer.access$0(ErsatzServer.groovy)
	at com.stehno.ersatz.ErsatzServer$1.handleRequest(ErsatzServer.groovy:401)
	at io.undertow.server.handlers.HttpTraceHandler.handleRequest(HttpTraceHandler.java:70)
	at io.undertow.security.handlers.AuthenticationCallHandler.handleRequest(AuthenticationCallHandler.java:52)
	at io.undertow.security.handlers.AuthenticationConstraintHandler.handleRequest(AuthenticationConstraintHandler.java:53)
	at io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
	at io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
	at io.undertow.server.handlers.encoding.EncodingHandler.handleRequest(EncodingHandler.java:72)
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:360)
	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:830)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
</pre>
</span>
</div>
<div id="tab2" class="tab">
<h2>Standard error</h2>
<span class="code">
<pre>Jul 20, 2018 6:32:02 AM okhttp3.internal.platform.Platform log
WARNING: previous digest authentication with same nonce failed, returning null
</pre>
</span>
</div>
</div>
<div id="footer">
<p>
<div>
<label class="hidden" id="label-for-line-wrapping-toggle" for="line-wrapping-toggle">Wrap lines
<input id="line-wrapping-toggle" type="checkbox" autocomplete="off"/>
</label>
</div>Generated by 
<a href="http://www.gradle.org">Gradle 4.8.1</a> at Jul 20, 2018 6:32:14 AM</p>
</div>
</div>
</body>
</html>
